mvlearn/decomposition/baseica.py

```diff
@@ -0,0 +1,233 @@
+# MIT License
+
+# Copyright (c) [2017] [Pierre Ablin and Hugo Richard]
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import numpy as np
+from scipy import linalg
+from sklearn.decomposition import PCA
+
+from .base import BaseDecomposer
+from ..preprocessing.repeat import ViewTransformer
+
+
+class BaseICA(BaseDecomposer):
+    """A base class for multiview ICA methods.
+
+    Parameters
+    ----------
+    n_components : int, optional
+        Number of components to extract. If None, n_components is set to
+        the minimum number of features in the dataset.
+
+    preproc: None, 'pca' or a ViewTransformer-like instance,\
+            default='pca'
+        Preprocessing method to use to reduce data.
+        If None, no preprocessing is applied.
+        If "pca", performs PCA separately on each view to reduce dimension
+        of each view.
+        Otherwise the preprocessing is performed using the transform
+        method of the ViewTransformer-like object ignoring the `n_components`
+        parameter. This instance also needs an inverse transform method
+        to recover original data from reduced data.
+
+    multiview_output : bool, default True
+        If True, the `.transform` method returns one dataset per view.
+        Otherwise, it returns one dataset, of shape (n_samples, n_components)
+
+    random_state : int, RandomState instance or None, default=None
+        Used to perform a random initialization. If int, random_state is
+        the seed used by the random number generator; If RandomState
+        instance, random_state is the random number generator; If
+        None, the random number generator is the RandomState instance
+        used by np.random.
+
+    verbose : bool, default=False
+        Print information
+
+    Attributes
+    ----------
+    preproc_instance : ViewTransformer-like instance
```

Comment:
only attribute names passed to the init are allowed to end without an underscore